src/Mono.Android/Java.Lang/Object.cs

```diff
@@ -14,64 +14,31 @@
 namespace Java.Lang {
 
 	[Serializable]
-	public partial class Object : IDisposable, IJavaObject, IJavaObjectEx
-#if JAVA_INTEROP
-		, IJavaPeerable
-#endif  // JAVA_INTEROP
+	public partial class Object : global::Java.Interop.JavaObject, IJavaObject, IJavaObjectEx
 	{
-		[NonSerialized] IntPtr key_handle;
-#pragma warning disable CS0649, CS0169, CS0414 // Suppress fields are never used warnings, these fields are used directly by monodroid-glue.cc
-		[NonSerialized] int refs_added;
-#pragma warning restore CS0649, CS0169, CS0414
-		[NonSerialized] JObjectRefType handle_type;
-		[NonSerialized] internal IntPtr handle;
-		[NonSerialized] bool             needsActivation;
-		[NonSerialized] bool             isProxy;
```

Comment:
Regarding the one failure related to JSON serialization, do we need `[NonSerialized]` on the fields here?

https://github.com/dotnet/java-interop/blob/c86ae26c5814766af6e1a191d94ac29c0eb6039b/src/Java.Interop/Java.Interop/JavaObject.cs#L19-L28

Reply:
Yup, done here: https://github.com/dotnet/java-interop/pull/1293/commits/9d1cae1ae74a9a9d36b0f3f528fe50bc3c2ee818